server/src/main/java/com/hedera/block/server/notifier/NotifierImpl.java

```diff
@@ -0,0 +1,158 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server.notifier;
+
+import static com.hedera.block.server.metrics.BlockNodeMetricTypes.Counter.SuccessfulPubStreamResp;
+import static com.hedera.block.server.metrics.BlockNodeMetricTypes.Gauge.Producers;
+import static com.hedera.block.server.producer.Util.getFakeHash;
+import static java.lang.System.Logger.Level.ERROR;
+
+import com.hedera.block.server.config.BlockNodeContext;
+import com.hedera.block.server.events.BlockNodeEventHandler;
+import com.hedera.block.server.events.ObjectEvent;
+import com.hedera.block.server.mediator.SubscriptionHandlerBase;
+import com.hedera.block.server.metrics.MetricsService;
+import com.hedera.block.server.service.ServiceStatus;
+import com.hedera.hapi.block.Acknowledgement;
+import com.hedera.hapi.block.EndOfStream;
+import com.hedera.hapi.block.ItemAcknowledgement;
+import com.hedera.hapi.block.PublishStreamResponse;
+import com.hedera.hapi.block.PublishStreamResponseCode;
+import com.hedera.hapi.block.stream.BlockItem;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.lmax.disruptor.BatchEventProcessor;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.security.NoSuchAlgorithmException;
+import java.util.Map;
+
+/**
+ * Use NotifierImpl to mediate the stream of responses from the persistence layer back to multiple
+ * producers.
+ *
+ * <p>As an implementation of the StreamMediator interface, it proxies block item persistence
+ * responses back to the producers as they arrive via a RingBuffer maintained in the base class and
+ * persists the block items to a store. It also notifies the mediator of critical system events and
+ * will stop the server in the event of an unrecoverable error.
+ */
+class NotifierImpl extends SubscriptionHandlerBase<PublishStreamResponse> implements Notifier {
+
+    private final System.Logger LOGGER = System.getLogger(getClass().getName());
+
+    private final Notifiable mediator;
+    private final MetricsService metricsService;
+    private final ServiceStatus serviceStatus;
+
+    NotifierImpl(
+            @NonNull
+                    final Map<
+                                    BlockNodeEventHandler<ObjectEvent<PublishStreamResponse>>,
+                                    BatchEventProcessor<ObjectEvent<PublishStreamResponse>>>
+                            subscribers,
+            @NonNull final Notifiable mediator,
+            @NonNull final BlockNodeContext blockNodeContext,
```

Comment:
I thought we'd decided to use BlockNodeContext as a wrapper object to future-proof against adding additional parameters. Dagger works for these classes but I spent a lot of time repairing the constructors in the tests because we're not using DI there.

Reply:
Yes, lets leave as it is. 👍